components/TokenDropdownGauges.tsx

```diff
@@ -0,0 +1,77 @@
+import React, {Fragment, ReactElement, cloneElement, useRef} from 'react';
+import {Menu, Transition} from '@headlessui/react';
+import IconChevron from 'components/icons/IconChevron';
+
+export type TDropdownGaugeOption = {
+	icon?: ReactElement;
+	label: string;
+	value: any;
+	zapVia: string;
+};
+
+export type TDropdownGaugesProps = {
+	options: TDropdownGaugeOption[];
+	defaultOption: TDropdownGaugeOption;
+	selected: TDropdownGaugeOption;
+	placeholder?: string;
+	onSelect:
+		| React.Dispatch<React.SetStateAction<TDropdownGaugeOption>>
+		| ((option: TDropdownGaugeOption) => void);
+};
+
+function DropdownGauges({options, defaultOption, selected, onSelect, placeholder = ''}: TDropdownGaugesProps): ReactElement {
+	const buttonRef = useRef<HTMLButtonElement>(null);
+	return (
+		<div>
+			<Menu as={'menu'} className={'relative inline-block w-full text-left'}>
+				{({open}): ReactElement => (
+					<>
+						<Menu.Button
+							ref={buttonRef}
+							className={'flex h-10 w-full items-center justify-between border border-neutral-600 p-2 text-base text-neutral-900'}>
+							<div className={'flex flex-row items-center'}>
+								{selected?.icon ? cloneElement(selected.icon) : <div className={'h-6 w-6 rounded-full bg-neutral-500'} />}
+								<p className={`pl-2 ${(!selected?.label && !defaultOption?.label) ? 'text-neutral-400' : 'text-neutral-900'} font-normal`}>
+									{selected?.label || defaultOption?.label || placeholder}
```

Comment:
```suggestion
									{selected?.label || defaultOption?.label || placeholder || 'Unknown'}
```
if everything else fails, having empty string might be confusing